server/container_create.go

```diff
@@ -163,6 +164,28 @@ func buildOCIProcessArgs(containerKubeConfig *pb.ContainerConfig, imageOCIConfig
 	return processArgs, nil
 }
 
+// addOCIHooks look for hooks programs installed in hookDirPath and add them to spec
+func addOCIHooks(specgen *generate.Generator, configFile string) error {
+	const hookDirPath = "/usr/libexec/oci/hooks.d"
+
+	// find any hooks executables
+	if _, err := os.Stat(hookDirPath); os.IsNotExist(err) {
+		return nil
+	}
+	hooks, err := ioutil.ReadDir(hookDirPath)
+	if err != nil {
+		return err
+	}
+	for _, item := range hooks {
+		if item.Mode().IsRegular() {
+			hook := filepath.Join(hookDirPath, item.Name())
+			specgen.AddPreStartHook(hook, []string{hook, "prestart", configFile})
```

Comment:
Some of the hooks look at the command to be executed inside of the container.  I have not fully tested if they work correctly.  In a docker container we are passing something slightly different, but I might need to cleanup the hooks.